code/modules/surgery/stomachpump.dm

```diff
@@ -44,6 +44,8 @@
 				"<span class='notice'>[user] forces [H] to vomit, cleansing their stomach of some chemicals!</span>",
 				"[user] forces [H] to vomit!")
 		H.vomit(20, FALSE, TRUE, 1, TRUE, FALSE, purge = TRUE) //called with purge as true to lose more reagents
+		if(HAS_TRAIT(target,TRAIT_SURGERYPLUS))
+			target.reagents?.remove_all(5) //additional purging that is irretrievable (in case someone had the crazy idea to make reagents extractable via vomit).
```

Comment:
can't you scoop up vomit pools off of the ground using a beaker?

Reply:
I cannot remember off the top of my head, but good thing I added this then lol